Formula/lfe.rb

```diff
@@ -13,18 +13,15 @@ class Lfe < Formula
     sha256 cellar: :any_skip_relocation, mojave:        "16d8e8a9f9ab091b1184dac366bb9d83cfccfdb728f79c035362b7b86940a7af"
   end
 
-  depends_on "emacs" if MacOS.version >= :catalina
```

Comment:
I think we can have it as a build dependency instead.

Reply:
But, it isn't required.

Reply:
It isn't required only if you don't include the `elisp` files as well, which is why you removed them from the `install` method. Instead, if you make it a build-only dependency, you get the `elisp` files installed, but you won't need `emacs` when you do `brew install lfe`.

Reply:
> Having it around doesn't make sense when we are installing LFE.

It is required for the `make "emacs"` step which is used to install the `elisp` files. And that's why I suggested adding it as a build-only dependency. This means that `emacs` will not be installed when you install `lfe` using `brew install lfe`.

Reply:
I still beg to differ on this suggestion. I'll give you two reasons:

Reason 1:
The reason the `elisp` file was added and hence, emacs dependency was added to homebrew, I think, is because lfe repo has those emacs files present in the repo to support lfe in emacs: https://github.com/lfe/lfe/tree/develop/emacs. 
Now, sure that would make sense if you didn't have emacs installed on your machine and wanted to work on lfe quickly but, that isn't the situation usually. If you use emacs, you will most likely have your emacs config and can easily pull those files into your config and make it work and like me, if you have emacs-plus installed it will interfere with it or if you have emacs already, why would you want it to be installed again? Making emacs unnecessary dependency.

Reason 2:
You mentioned, having a build-only dependency. Having it as a build-only dependency totally ignores the purpose of its presence. Like I mentioned in the first line in reason 1, it is present to provide lfe support in an emacs.

Reply:
The purpose of the `emacs` dependency is so that we can build the `*.elc` files in case a user would like to make use of them. Perhaps there is another way a user can acquire these files if they want to, but having to figure out that they need to get them elsewhere (especially given that the formula has provided these files for years) makes for a terrible user experience.

Skipping the `make emacs` and `elisp.install` steps is definitely out of the question here.

Reply:
Quick caveat: I'm not pushing for one solution over another, just trying to understand the parameters here ;-)

@carlocab Curious as to why that's out of the question? Erlang also comes with Emacs files ... I don't see emacs as a dependency of that project? I'm looking here (maybe I missed it):

* https://github.com/Homebrew/homebrew-core/blob/137b3bf8376c8e48f4d250b815446f06d9067343/Formula/erlang.rb

Reply:
(I can imagine a couple of reasons why Homebrew wouldn't want to make this change, but won't muddy the waters with my imaginings :-D -- best to hear directly from the project maintainers!)

Reply:
@carlocab okay, here's the logic in my head (and why the hard stance against  removing an unnecessary dep is a mystery):

1.  The original contribution to Homebrew for LFE was overzealous -- we expect that most of our users do not install the Emacs files.
2. LFE is a dialect of Erlang (technically it's a simple Erlang library); it should have the same dependencies as Erlang itself -- no more, no less.
3. Erlang also provides Emacs files that the user may install at their convenience -- the Erlang Homebrew formuala does not do that for the Erlang install.
4. The symmetry with Erlang is broken in the LFE Homebrew formula, and has been causing inconvenience (and sometimes problems) for users almost as long as the LFE project has been a part of Homebrew.

As such, it seems a natural thing to remove the `make emacs` step thus bringing it into symmetry with the Erlang formula. 

What is the specific reason for the hard "no" to bringing the LFE install in line with the Erlang install?

Reply:
@carlocab:

> I explain why in the first part of my previous comment. 

Right, you said: 

> The purpose of the emacs dependency is so that we can build the *.elc files in case a user would like to make use of them. 

That's the literal reason why the build steps is there -- I fully understand that. That was never a mystery ;-)

What I'm wondering is: 

1. if we're telling you it's not necessary for this project and
2. the Erlang Homebrew formula does not perform this build step,
3. why won't you consider removing it? what am I missing?

Reply:
> 1. we expect most of our users to not install the Emacs files.

If that was the expectation, I would've expected a note in the Makefile or README. Of course, I may have missed this recommendation/expectation elsewhere if it exists.

> 3\. why won't you consider removing it? what am I missing?

I believe Carlo's comment made that clear: the `*.elc` files have been installed with the formula for years, and removing them all of a sudden makes for a bad user experience.

> 2\. The emacs dependency was added to the formula at a later date by someone who was not a core LFE contributor

We cannot expect only core contributors of an upstream project to make changes to the formula. A user requested this feature to be included and a Homebrew maintainer approved those changes, as they believed them to be reasonable/useful.

I hear your points and I want this formula to be useful (to everyone) and obviously not broken. However I don't get why there is insistence to remove something we've offered for years, without much clear explanation as to why this _breaks_ the formula or renders it _unusable_. Note that we cannot make changes in `homebrew/core` for compatibility with third-party Taps.

It would really help us a lot more if you could explain what's breaking and whether including the `*.elc` files is the cause of the problem. I don't think it is, which is why I suggested making `emacs` a build-only dependency. This would really solve the original problem of `emacs` being installed when you have your preferred (non `homebrew/core`) variant of `emacs` installed.

Reply:
Okay, the user experience was my actual guess -- thanks for clarifying (and confirming).

It's too bad the change in 2016 was made without oversight from active members of the LFE community, as that has caused problems for many. Yeah, you're correct: the conflict with that specific installed emacs is what's causing the problems. To be honest, the end result has been that the more active LFE users (the ones we see most in Slack) don't use Homebrew for LFE.

@prertik -- I'll leave this ticket to you now, however you want to proceed with regard to the build-only use of Homebrew `emacs` ...

Thanks again, @nandahkrishna.

Reply:
As for your point about `erlang` not including these: to my knowledge, no PR or issue has been opened for their inclusion. If someone wishes to add it in the future, they would be welcome to. At that juncture of course, the pros and cons or any objections would be evaluated. Again, I don't see why including the `elisp` files is the root of your problem here.

Reply:
I think @nandahkrishna covers a lot of what I wanted to say, but just re-emphasise certain things:

A lot of the reasons you give above just aren't relevant considerations for us. (E.g. symmetry between formulae, the status of Erlang's dependency tree and what that entails about LFE's dependency tree, or whether a change to the formula was made by a core contributor or not.)

Not all of your reasons are irrelevant -- we do care that there are users who are unhappy about the state of the formula, and we do want to try to make things better for them. We want to do so in a way that doesn't simultaneous remove features from a formulae that have been there for years, especially not when those features have gotten almost no complaints until now.

I don't really have the time, patience, or inclination to carry on an extended discussion about this, so let me just make the following clear: 

Please make emacs a build dependency and restore the install steps that use it. Otherwise, we are respectfully declining this pull request. Thanks.

Reply:
Also just to respond to some of @prertik's points (in https://github.com/Homebrew/homebrew-core/pull/79032#discussion_r648524117) about my suggestion:
* I think your *Reason 1* does not take into account the fact that build-dependencies won't be installed with the formula when doing the standard `brew install`. As for easily pulling the files, Homebrew users expect us to do the pulling for them. It may be easy to do it independently for other advanced users such as yourself. These files aren't overriding your Emacs configuration either, so I fail to see the problem with including them for those who wish to use them. We want to be _useful_ for everyone.
* As for *Reason 2*, yes it exists to provide additional features to use in conjunction with `emacs`. And so they will be used by anyone who has `emacs` or any variant installed, and wishes to use them. I don't see how making it a build dependency renders these files useless – clearly people install variants of Emacs from outside of `homebrew/core`, and there are users wishing to use these our at least not bothered by these, as they've been provided for years with no complaint.